src/components/pill/pill.style.js

```diff
@@ -0,0 +1,35 @@
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import baseTheme from '../../style/themes/base';
+import classicThemeForPill from './pill-classic.style';
+import modernThemeForPill from './pill-modern.style';
+import { THEMES } from '../../style/themes';
+import OptionsHelper from '../../utils/helpers/options-helper';
+
+const PillStyle = styled.span`
+  ${props => (isClassic(props) ? classicThemeForPill : modernThemeForPill)}
+`;
+
+function isClassic({ theme, styledAs }) {
+  // handles incorrect default activeTheme in StoryBook AppWrapper
```

Comment:
This shouldn't really be in the component itself, it should be part of the story book usage since this is a specific issue with the StoryBook AppWrapper, rather than the Pill component itself.

Reply:
Thanks Chris, I agree but because the App Layout determines the theme and wraps the Pill with a Theme Provider before it renders the component it's already set before we try to style it. We know that there's a requirement for some middleware to better handle this, but for now this code is an unobtrusive workaround. When the middleware is available we'll revisit this and correct it.

Reply:
@chrisbarber86 are you satisfied with this?